From 6bb8872372f950632baed78dc74a62b888e83845 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Fri, 20 Apr 2018 11:39:34 +0300
Subject: [PATCH 131/706] enetc: Enable Rx VLAN filtering by setting
 PSIVLANFMR[VS]

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 47156d9405bbb5d9d2644c486bd2ab5cb9f1b32b)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc_hw.h | 3 +++
 drivers/net/ethernet/freescale/enetc/enetc_pf.c | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc_hw.h b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
index 0da594cce33e..858cc9689b58 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_hw.h
+++ b/drivers/net/ethernet/freescale/enetc/enetc_hw.h
@@ -195,6 +195,9 @@ enum enetc_bdr_type {TX, RX};
 #define ENETC_RSSHASH_KEY_SIZE	40
 #define ENETC_PRSSK(n)		(0x01410 + (n) * 4) /* n = [0..9] */
 
+#define ENETC_PSIVLANFMR	0x01700
+#define ENETC_PSIVLANFMR_VS	BIT(0)
+
 #define ENETC_PRFSMR		0x01800
 #define ENETC_PRFSMR_RFSE	BIT(31)
 #define ENETC_PRFSCAPR		0x01804
diff --git a/drivers/net/ethernet/freescale/enetc/enetc_pf.c b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
index 8fca8040ad0e..542ca7c6e112 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc_pf.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc_pf.c
@@ -491,6 +491,8 @@ static void enetc_port_si_configure(struct enetc_si *si)
 	/* Port level VLAN settings */
 	val = ENETC_PVCLCTR_OVTPIDL(ENETC_VLAN_TYPE_C | ENETC_VLAN_TYPE_S);
 	enetc_port_wr(hw, ENETC_PVCLCTR, val);
+	/* use outer tag for VLAN filtering */
+	enetc_port_wr(hw, ENETC_PSIVLANFMR, ENETC_PSIVLANFMR_VS);
 }
 
 static void enetc_configure_port_mac(struct enetc_hw *hw)
-- 
2.17.1

